### ISSUE TITLE
`@SuppressWarnings("UnusedException")` does not suppress the error

### ISSUE THREAD

**Issue by dpursehouse** [google/error-prone#1484]
> ATTENTION! Please read and follow:
> - if this is a _question_ about Error Prone, send it to error-prone-discuss@googlegroups.com
> - if this is a _bug_ or _feature request_, fill the form below as best as you can.

### Description of the problem / feature request:

Code that intentionally catches an exception and then throws another exception, without setting the original exception as cause.  Add the suppression as described in the [documentation](https://errorprone.info/bugpattern/UnusedException), but the error is still raised.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

The following code is from `LfsPointer` in the jgit project:

```
	public void encode(OutputStream out) {
		try (PrintStream ps = new PrintStream(out, false,
				UTF_8.name())) {
			ps.print("version "); //$NON-NLS-1$
			ps.print(VERSION + "\n"); //$NON-NLS-1$
			ps.print("oid " + HASH_FUNCTION_NAME + ":"); //$NON-NLS-1$ //$NON-NLS-2$
			ps.print(oid.name() + "\n"); //$NON-NLS-1$
			ps.print("size "); //$NON-NLS-1$
			ps.print(size + "\n"); //$NON-NLS-1$
		} catch (@SuppressWarnings("UnusedException") UnsupportedEncodingException e) {
			// should not happen, we are using a standard charset
			throw new UnsupportedCharsetException(UTF_8.name());
		}
	}
```

If I remove the `throw new ...` line, then there's no problem, but I would expect the suppression to work as described.

### What version of Error Prone are you using?

Whatever version is used in bazel 2.0.0

###  Have you found anything relevant by searching the web?

No

**Comment by dpursehouse** [comment]
I've found that the suppression works if I add it on the method rather than only on the exception parameter, so maybe this is just a documentation problem.

**Comment by Stephan202** [comment]
IIUC this is a duplicate of #1156. That issue is fixed as of Error Prone 2.3.3. (The latest release at this time is 2.3.4.)

**Comment by dpursehouse** [comment]
Thanks @Stephan202 it looks like you're right.  I'll close this.
